### PR TITLE
status: Add CredentialsRequests to relatedObjects

### PR DESCRIPTION
**Pull request by Miciah**
Add all CredentialRequests resources to `relatedObjects` so that [must-gather](https://github.com/openshift/must-gather) will gather them.

* `pkg/controller/credentialsrequest/status.go` (`syncOperatorStatus`): Pass the list of CredentialRequests to `buildExpectedRelatedObjects`.
(`buildExpectedRelatedObjects`): Add a parameter for a slice of CredentialRequests.  Add the provided CredentialRequests resources the list of related objects.

### CODE REVIEW

File: pkg/controller/credentialsrequest/status.go [hunk 1]
```diff
@@ -276,11 +276,20 @@ func findClusterOperatorCondition(conditions []configv1.ClusterOperatorStatusCon
...
+	related := []configv1.ObjectReference{
 		{
 			Resource: "namespaces",
 			Name:     cloudCredOperatorNamespace,
 		},
 	}
+	for _, cr := range credRequests {
+		related = append(related, configv1.ObjectReference{
+			Group:     minterv1.SchemeGroupVersion.Group,
+			Resource:  "CredentialsRequest",
```
Comment (abhinavdahiya) [inline pkg/controller/credentialsrequest/status.go]: the resource is wrong, this should be `credentialsrequests`

Comment (abhinavdahiya) [reply pkg/controller/credentialsrequest/status.go]: https://github.com/openshift/cloud-credential-operator/blob/77eaf88b08645f1953b657bf0109aab15054c50b/manifests/00_v1_crd.yaml#L12

Comment (Miciah) [reply pkg/controller/credentialsrequest/status.go]: How is the `plural` field more correct to use here than the `kind` field?  Is there an API that using kind violates, or a tool that it breaks?  For what it's worth, I tested this change and verified that must-gather gathered the CredentialsRequest resources.  Morever, use the kind (upper camel case) in the ingress operator too have not seen or received reports of any problems.

**Pull request merged**